src/Http/Http.Abstractions/src/Extensions/HttpResponseWritingExtensions.cs

```diff
@@ -126,7 +139,7 @@ private static void WriteMultiSegmentEncoded(PipeWriter writer, string text, Enc
                 writer.Advance(bytesUsed);
                 source = source.Slice(charsUsed);
 
-                destination = writer.GetSpan(encodedLength - totalBytesUsed);
+                destination = writer.GetSpan(minimumByteSize);
```

Comment:
Can minimumByteSize ever be greater than `encodedLength - totalBytesUsed`?

Reply:
Never mind. I don't know why I though this was a problem. Totally makes sense why you use the smallest value possible. Hopefully it doesn't come back to bite us.